source/base/fileinputoutput.cpp

```diff
@@ -317,12 +319,19 @@ IStream *NewIStream(const Path& p, unsigned int stype)
         throw POV_EXCEPTION(kCannotOpenFileErr, str);
     }
 
-    return new IFileStream(p().c_str());
+    stream = new IFileStream(p().c_str());
+    if (!(*stream))
+    {
+        delete stream;
+        return NULL;
+    }
+    return stream;
```

Comment:
I had chosen this method after finding that much (if not all) of the code downstream from these functions uses comparison to NULL to detect failure (and throw an exception); e.g.:

* source/base/image/image.cpp:3943 (via ????)
* source/backend/scene/backendscenedata.cpp:292 (via NewOStream)
* source/backend/scene/backendscenedata.cpp:303 (via NewOStream)
* source/frontend/imageprocessing.cpp:181 (via NewOStream)
* source/frontend/processoptions.cpp:81 (via OpenFileForRead)
* source/frontend/processoptions.cpp:373 (via OpenFileForWrite)
* source/parser/parser.cpp:6499 (via Locate_File)
* source/parser/parser.cpp:10937 (via ReadFile)
* source/parser/parser_materials.cpp:1443 (via Locate_File)
* source/parser/parser_tokenizer.cpp:2608 (via Locate_File)
...

I'm not entirely sure, but it seems that such code may have been relying on the cast from `IOBase` to `const void *` (in analogy to the standard C++ I/O); unfortunately, it uses `IOBase *`, which doesn't work the same.  I chose to modify NewIStream / NewOStream since all (as far as I can tell) of such 'confused'(?) code passes through those functions, and this would restore the expected function.

I'd be willing to try to convert things to use exceptions directly (or whatever), but it may take some time.

Let me know.